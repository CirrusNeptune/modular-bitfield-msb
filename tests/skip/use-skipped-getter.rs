use modular_bitfield::prelude::*;

#[bitfield]
#[derive(Debug)]
pub struct Sparse {
    #[skip(getters)]
    unused_1: B10,
    a: bool,
    #[skip(getters)]
    unused_2: B10,
    b: bool,
    #[skip(getters)]
    unused_3: B10,
}

fn main() {
    let sparse = Sparse::from_bytes([0xFF; 4]);
    sparse.set_unused_1(0);
    assert_eq!(sparse.unused_1(), 0); // ERROR!
}
