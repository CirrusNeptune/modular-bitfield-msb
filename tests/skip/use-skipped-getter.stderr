error[E0599]: no method named `unused_1` found for struct `Sparse` in the current scope
  --> $DIR/use-skipped-getter.rs:19:23
   |
4  | / #[derive(Debug)]
5  | | pub struct Sparse {
   | |___- method `unused_1` not found for this
...
19 |       assert_eq!(sparse.unused_1(), 0); // ERROR!
   |                         ^^^^^^^^ method not found in `Sparse`
